### PR TITLE
Adapt shared lock skip logic to dbswap branch

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: internal/cmd/commands/server/server.go [hunk 1]
```diff
@@ -887,7 +894,7 @@ func (c *Command) reloadControllerDatabase(newConfig *config.Config) error {
 	}
 
 	// Acquire new lock on the new database and verify that it's in a good state to be used.
-	newDbSchemaManager, err := acquireDatabaseSharedLock(c.Context, newDb)
+	newDbSchemaManager, err := acquireSchemaManager(c.Context, newDb, c.Config.Controller.Database.SkipSharedLockAcquisition)
```
Comment (hugoamvieira) [inline internal/cmd/commands/server/server.go:897]:
```
if c.Config.Controller.Database.SkipSharedLockAcquisition {
			if err := c.schemaManager.Close(c.Context); err != nil {
				c.UI.Error(fmt.Errorf("Unable to release shared lock to the database: %w", err).Error())
				return base.CommandCliError
			}
		}
```

We need to do this for `newDbSchemaManager` too after we verify its state

Comment (jefferai) [reply internal/cmd/commands/server/server.go:897]: Good catch!

File: internal/db/schema/manager.go [hunk 2]
```diff
@@ -6,13 +6,16 @@ import (
...
+	"sync"
 
 	"github.com/hashicorp/boundary/internal/db/schema/internal/edition"
 	"github.com/hashicorp/boundary/internal/db/schema/internal/log"
 	"github.com/hashicorp/boundary/internal/db/schema/internal/postgres"
 	"github.com/hashicorp/boundary/internal/db/schema/internal/provider"
 	"github.com/hashicorp/boundary/internal/db/schema/migration"
 	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/observability/event"
+	"go.uber.org/atomic"
```
[the remark this answers is not in the document]
Comment (jefferai) [reply internal/db/schema/manager.go:18]: ah no I meant to use the new stdlib stuff...either is honestly fine but no reason to use this, will fix!

**Review by tmessi — APPROVED** [review]

**Review by hugoamvieira — APPROVED** [review]
Did a bunch more testing this morning, all looks great, thank you!

**Pull request merged**